packages/openid4vc/src/shared/utils.ts

```diff
@@ -66,23 +67,29 @@ export function getVerifyJwtCallback(agentContext: AgentContext): VerifyJwtCallb
   }
 }
 
-export function getCreateJwtCallback(agentContext: AgentContext): CreateJwtCallback {
+export function getCreateJwtCallback(
+  agentContext: AgentContext
+): CreateJwtCallback<DPoPJwtIssuerWithContext | VpJwtIssuerWithContext> {
   return async (jwtIssuer, jwt) => {
     const jwsService = agentContext.dependencyManager.resolve(JwsService)
 
     if (jwtIssuer.method === 'did') {
       const key = await getKeyFromDid(agentContext, jwtIssuer.didUrl)
       const jws = await jwsService.createJwsCompact(agentContext, {
-        protectedHeaderOptions: { alg: jwtIssuer.alg, ...jwt.header },
+        protectedHeaderOptions: { ...jwt.header, alg: jwtIssuer.alg, jwk: undefined },
         payload: JwtPayload.fromJson(jwt.payload),
         key,
       })
 
       return jws
     } else if (jwtIssuer.method === 'jwk') {
-      const key = getJwkFromJson(jwtIssuer.jwk).key
+      if (!jwtIssuer.jwk.kty) {
+        throw new CredoError('Missing required key type (kty) in the jwk.')
+      }
```

Comment:
again, why this check?

Reply:
can be undefined in sphereon

Reply:
But in this case we have a jwk class instance right? So it won't happen?